### ISSUE TITLE
TextArea: Screen reader doesn't announce helper message change

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2112]
**Describe the bug**
Screen reader doesn't announce helper message change for max length (# characters left) like it does for input element. The reason is because we doesn't add aria-describedby attribute when we have only maxCount props for textarea.

**To Reproduce**
1. Go to 'https://react-magma.cengage.com/version/4.10.1/api/textarea/#character_counter'
2. Turn on NVDA or VO.
3. Type text in the text area
4. NVDA and VO announce only the typed text, not the helper message.

**Expected behavior**
 NVDA and VO should announce the helper message when the user changes the text area value.

**Screenshots**

<img width="1577" height="729" alt="Image" src="https://github.com/user-attachments/assets/2b894515-b86a-4d7c-9662-b9a4145cb9eb" />

**Additional context**
[Slack request](https://cengage.slack.com/archives/C8Y4FR63U/p1764935929079959)

**Comment by silvalaura** [comment]
Could be similar to https://github.com/cengage/react-magma/issues/1124
